pandas/core/internals/api.py

```diff
@@ -0,0 +1,61 @@
+"""
+This is a pseudo-public API for downstream libraries.  We ask that downstream
```

Comment:
I would have these import to pandas.api.internals i think (then you can *really* control the exports)

Reply:
i know pyarrow accesses the pd.core.internals namespace.  not sure about others.  we can ask them to change, but for the forseeable future will need these in the namespace.

Reply:
right what i  mean is let's expose a wrapper api namespace and then we can change the downstream packages when we have released.

Reply:
fair enough.  keep it in this file though?  im wary of adding it to the pd.api namespace lest _new_ downstream packages adopt bad habits

Reply:
ok sure